compiler/rustc_middle/src/query/keys.rs

```diff
@@ -125,56 +133,67 @@ impl<'tcx> Key for mir::interpret::LitToConstInput<'tcx> {
 
 impl Key for CrateNum {
     type CacheSelector = VecCacheSelector<Self>;
+    type LocalKey = ();
```

Comment:
I would prefer some kind of `struct LocalCrate;`. That makes it more obvious that it's just a local crate provider and not a unit query. (The providers would use `|tcx, LocalCrate| { ... }`)